src/hal/src/adapter.rs

```diff
@@ -43,7 +45,7 @@ pub struct MemoryProperties {
 }
 
 /// Represents a physical or virtual device, which is capable of running the backend.
-pub trait PhysicalDevice<B: Backend>: Sized {
+pub trait PhysicalDevice<B: Backend>: Any + Send + Sync {
```

Comment:
why do we need `Any` here?

Reply:
Just for the case the user would want to handle `PhysicalDevice` from different backends differently. It is more convenient to cast `PhysicalDevice` instead of matching typeid of `Backend`.

Reply:
I'm fine with physical devides having `Any` as they are somewhat top level (just above the instance) and useful without knowledge of specific backend. But I don't think `Any` helps us on other types down the line...